src/CacheManager.Microsoft.Extensions.Caching.Memory/MemoryCacheHandle`1.cs

```diff
@@ -87,6 +89,26 @@ public override bool Exists(string key, string region)
             return _cache.Contains(GetItemKey(key, region));
         }
 
+        /// <inheritdoc />
+        public override IEnumerable<string> Keys(string pattern, string region)
+        {
+            var keys = _cache.ListChildren(region ?? GetType().Name)
+                .Select(k => ParseKey(region));
+
+            if (pattern == "*")
+            {
+                return keys;
+            }
+            if (pattern.Contains("*") || pattern.Contains("?"))
```

Comment:
looks like old matching implementation not using the extension method? Is there any reason for that?